lib/searches.js

```diff
@@ -179,48 +160,124 @@ let dfsHelper = function (url, edges, level) {
+let depthFS = function (rootUrl, searchDepth, cookie) {
     let currentDepth = 0;
     let edges = [];
     let nextUrl = rootUrl;
+    let linkError = false;
 
     let searchingPromise = new Promise((resolve, reject) => {
         console.log("Crawling level 0...");
 
         // Get random links until the searchDepth is reached.
         function getRandomLink(url) {
+            
+            // If searchDepth is reached, insert the tree into the database
             if (currentDepth >= searchDepth) {
-                resolve(edges);
+                resolve(1);
                 return;
-            }
-
+            } 
+            
+            // Call dfsHelper to insert the next random link into the edge array
             let helperPromise = dfsHelper(nextUrl, edges, currentDepth);
 
             helperPromise.then(() => {
                 // If link error, return results thus far.
                 if (edges[currentDepth] === undefined){
-                    resolve(edges);
+                    resolve(-1);
+                    return;
                 }
+               
+                if (edges[currentDepth].termFound === true){
+                    resolve(2);
+                    return;
+                }
+                // Insert link into temp edges table
+                let tempInsert = dbAPI.insertIntoTempEdges(edges[currentDepth].SourceUrl, edges[currentDepth].DestinationUrl, cookie, 'Ian Dalrymple', 'DFS', searchDepth, rootUrl);   
+               
+                tempInsert.then((result) => {});
+                tempInsert.catch((err) =>  {
+                    throw err;
+                });
+               
                 nextUrl = edges[currentDepth].DestinationUrl;
                 currentDepth++;
                 console.log("Crawling level " + currentDepth + "...");
+                
                 // Get another random link recursively.
                 getRandomLink(nextUrl);
             });
 
             helperPromise.catch((err) => {
-                resolve(edges);
+                reject(err);
                 return;
             });
         }
-
+        
         getRandomLink(nextUrl);
     });
 
     return searchingPromise;
 };
 
+// Adds search tree to the database.
+let addToDB = function(returnVal, url, cookie, searchType, searchDepth) {
+    let dbPromise = new Promise((resolve, reject) => {
+        let treeInsert = dbAPI.insertNewTree(url, 'Ian Dalrymple', cookie, searchType, searchDepth);
+                
+        treeInsert.then((result) => {
+            resolve(returnVal);
+            return;
+        });
+        treeInsert.catch((err) => {
+            throw err;
+        });
+    });
+    
+    return dbPromise;
+};
+
+// Returns 1 if search is already cached, routes to proper search function if
+// crawler needs to be run. 
+let crawl = function(searchType, rootURL, searchDepth, cookie, searchExists){
+    let crawlPromise = new Promise((resolve, reject) => {
```

Comment:
Like above, we don't need to create a new promise, we can leverage the promises returned from `breadthFS` and `depthFS`. We would need to make resolved promises for the cases where we don't call those methods, but there's a static helper method to make that easy.
```js
if (searchExists) {
  return Promise.resolve(1);
} else if (searchType === "BFS") {
  return breadthFS(rootUrl, searchDepth, cookie).then((result) => {
    return addToDB(result, rootUrl, cookie, "BFS", searchDepth);
  });
} else if (searchType === "DFS") {
  return depthFS(rootUrl, searchDepth, cookie).then((result) => {
    return addToDB(result, rootUrl, cookie, "DFS", searchDepth);
  });
} else {
  return Promise.resolve(-1);
}
```

Reply:
You're doing a good job with the Promises, I'm just offering suggestions to help you reuse what you've already done.